modules/flowable-cmmn-rest/src/test/java/org/flowable/cmmn/rest/service/api/runtime/VariableInstanceCollectionResourceTest.java

```diff
@@ -103,7 +104,32 @@ public void testQueryVariableExcludeLocalVariable() throws Exception {
 
     }
 
-    protected void assertResultsPresentInDataResponse(String url, int numberOfResultsExpected, String variableName, Object variableValue)
+    @CmmnDeployment(resources = { "org/flowable/cmmn/rest/service/api/repository/oneHumanTaskCase.cmmn" })
+    public void testVariableInstanceScopeIsPresent() throws Exception {
+        CaseInstance caseInstance = runtimeService.createCaseInstanceBuilder()
+                .caseDefinitionKey("oneHumanTaskCase")
+                .variable("myVar", "test1")
+                .start();
+
+        PlanItemInstance planItemInstance = runtimeService.createPlanItemInstanceQuery().planItemDefinitionId("theTask")
+                .caseInstanceId(caseInstance.getId()).singleResult();
+
+        Task task = taskService.createTaskQuery().caseInstanceId(caseInstance.getId()).singleResult();
+        taskService.setVariableLocal(task.getId(), "localTaskVariable", "localTaskVarValue");
+
+        runtimeService.setLocalVariable(planItemInstance.getId(), "myLocalVar", "test2");
+
+        String url = CmmnRestUrls.createRelativeResourceUrl(CmmnRestUrls.URL_VARIABLE_INSTANCES);
+
+        JsonNode node = assertResultsPresentInDataResponse(url + "?caseInstanceId=" + caseInstance.getId(), 3, "myLocalVar", "test2");
+        assertThat(node.path("scope").asText()).isEqualToIgnoringCase(RestVariable.RestVariableScope.LOCAL.name());
+        node = assertResultsPresentInDataResponse(url + "?caseInstanceId=" + caseInstance.getId() + "&excludeLocalVariables=true", 1, "myVar",
+                "test1");
+        assertThat(node.path("scope").asText()).isEqualToIgnoringCase(RestVariable.RestVariableScope.GLOBAL.name());
```

Comment:
Instead of doing the `assertResultsPresentInDataResponse` I think that the test would be clearer if we get the response and do `assertThatJson` on it instead.